src/AddControlDialog.cpp

```diff
@@ -0,0 +1,66 @@
+#include "ControlDialog.h"
+#include <QStandardItemModel>
+
+struct ControlRef {
+    QString name;
+    QString command;
+    QString icon;
+};
+
+std::map<QString, ControlRef> controls;
+
+ControlDialog::ControlDialog(QWidget* parent)
+	: QDialog(parent)
+{
+	setupUi(this);
+
+    //tclFunction->setVisible(false);
+    //connect(cbTclFunction, &QCheckBox::stateChanged, this, &ControlDialog::hasFunction);
+    connect(btnAdd, &QPushButton::clicked, this, &ControlDialog::onAddButtonClicked);
+    connect(btnRemove, &QPushButton::clicked, this, &ControlDialog::onRemoveButtonClicked);
+}
+
+void ControlDialog::hasFunction(int state)
+{
+    //tclFunction->setVisible(state == Qt::Checked);
+}
+
+void ControlDialog::setData(QString icon = {}, QString name = {}, QString command = {}, QString function = {})
```

Comment:
I know this code is still in development, and maybe you already changed this in later commits in this PR.

This is unusual: the `= {}` should be done in the .h file, not in the .cpp file.